code/game/objects/items/flag.dm

```diff
@@ -10,10 +13,25 @@
 		user.visible_message("<span class='notice'>[user] lights the [name] with [W].</span>")
 		fire_act()
 
+/obj/item/flag/fire_act(global_overlay = FALSE)
+	..()
+	update_icons()
+
+/obj/item/flag/extinguish()
+	..()
+	update_icons()
+
 /obj/item/flag/proc/update_icons()
-	overlays = null
-	overlays += image('icons/obj/flag.dmi', src , "fire")
-	item_state = "[icon_state]_fire"
+	if(burn_state == ON_FIRE)
+		overlays += image('icons/obj/flag.dmi', src , "fire")
```

Comment:
Flags that are held in-hand and have no "on fire" sprite do not show as burning when held